src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -2366,6 +2367,9 @@ protected virtual void MarkReflectionLikeDependencies (MethodBody body)
 			if (HasManuallyTrackedDependency (body))
 				return;
 
+			var scanner = new ReflectionMethodBodyScanner (this);
+			scanner.Scan (body);
```

Comment:
I'd like to see some numbers as the implementation look very allocation heavy and wasteful

Reply:
I'll add perf measurements to the running list of branch exit criteria. It doesn't make sense to measure now since we can definitely improve around the `StackSlot`.

Even with the inefficient `StackSlot` class and a lot more tracking (`UnknownValue.Instance` doesn't exist in .NET Native - it actually modelled the the type of the value all the time, even if it wasn't anything interesting because the scanner served other purposes too) in .NET Native, this didn't actually show up as an issue. The traces were dominated by the inefficient way CCI models IL instructions. Cecil has a similarly bad representation. I guess everyone took inspiration from Reflection.Emit.